### PR TITLE
Log replays from simulated tests and upload them as build artifacts in CI

### LINKED ISSUE #2050: Log replays during simulated tests and make them available as build artifacts in CI
### Description of the task

Log replays during simulated tests and make them available as build artifacts in CI

### Acceptance criteria

- [ ] Item A
- [ ] Item B
- [ ] Item C

### PR DESCRIPTION
**Pull request by jordanzliu**
### Description

Log replays from simulated tests and upload them as build artifacts in Github CI

- Replays are logged to the TEST_UNDECLARED_OUTPUTS_DIR enviroment variable location provided by Bazel
- The test outputs are put in .zip's by Bazel in `ai/hl/stp/whatever_play/test.outputs/outputs.zip` for each test
- ALL OF the test outputs end up in the same `.zip` thanks to Github CI

### Testing Done

Check CI for build artifacts

### Resolved Issues

Resolves #2050 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/simulated_tests/simulated_test_fixture.cpp [hunk 1]
```diff
@@ -57,6 +61,7 @@ void SimulatedTestFixture::SetUp()
     {
         enableVisualizer();
     }
+    setupReplayLogging();
```
Comment (akhilveeraghanta) [inline src/software/simulated_tests/simulated_test_fixture.cpp:64]: what happens when u run this locally? Could it use up a lot of storage over time?

**Review by akhilveeraghanta — COMMENTED** [review]
very impressive work, the minimal change is really an indicator of really good design on the protologging stuff, nice job as always :fire: :100: 

I only had 1 comment

File: src/software/simulated_tests/simulated_test_fixture.cpp (hunk 1 above)
Comment (jordanzliu) [reply src/software/simulated_tests/simulated_test_fixture.cpp:64]: Disk use is pretty insignificant, looks to be around 2 MB total for running all of the simulated tests.

Comment (akhilveeraghanta) [reply src/software/simulated_tests/simulated_test_fixture.cpp:64]: Ah okay, we can clean up the cache later if its piling up somehow, not sure how bazel cleans its test artifacts

**Review by akhilveeraghanta — DISMISSED** [review]
:fire:

**Review by itsarune — DISMISSED** [review]
Nothing from me 👌

File: src/software/simulated_tests/simulated_test_fixture.h [hunk 2]
```diff
@@ -39,6 +40,13 @@ class SimulatedTestFixture : public ::testing::Test
      */
     void enableVisualizer();
 
+    /**
+     * Creates a directory to output logs to in the directory at the
+     * TEST_UNDECLARED_OUTPUTS_DIR Bazel environment variable, and sets up some
```
Comment (jonathanlew) [inline src/software/simulated_tests/simulated_test_fixture.h:45]: it's worth linking https://docs.bazel.build/versions/master/test-encyclopedia.html#initial-conditions here

**Review by jonathanlew — COMMENTED** [review]
looks good

File: src/software/simulated_tests/simulated_test_fixture.h (hunk 2 above)
Comment (jordanzliu) [reply src/software/simulated_tests/simulated_test_fixture.h:45]: added

**Review by jonathanlew — APPROVED** [review]
LGTM!

**Review by akhilveeraghanta — APPROVED** [review]

**Review by akhilveeraghanta — APPROVED** [review]

**Pull request merged**